src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodCallTranslator.cs

```diff
@@ -0,0 +1,401 @@
+        /// </returns>
+        [CanBeNull]
+        Expression ArrayStaticHandler([NotNull] MethodCallExpression expression)
+        {
+            if (!expression.Method.IsStatic || expression.Method.DeclaringType != typeof(Array))
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case nameof(Array.IndexOf) when VersionAtLeast(9, 5) && expression.Method.GetParameters().Length <= 3:
+                return Coalesce(new SqlFunctionExpression("array_position", typeof(int), expression.Arguments.Take(3)));
+
+            default:
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Translates instance methods defined on <see cref="Array"/>.
+        /// </summary>
+        /// <param name="expression">The expression to translate.</param>
+        /// <returns>
+        /// A translated expression or null if no translation is supported.
+        /// </returns>
+        [CanBeNull]
+        Expression ArrayInstanceHandler([NotNull] MethodCallExpression expression)
+        {
+            if (!(expression.Object is Expression instance))
+                return null;
+
+            if (!instance.Type.IsArray)
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case nameof(Array.GetLength) when VersionAtLeast(8, 4):
+                return
+                    new SqlFunctionExpression(
+                        "array_length",
+                        typeof(int),
+                        new[] { instance, GenerateOneBasedIndexExpression(expression.Arguments[0]) });
+
+            case nameof(Array.GetLowerBound) when VersionAtLeast(7, 4):
+                return
+                    Coalesce(
+                        new SqlFunctionExpression(
+                            "array_lower",
+                            typeof(int),
+                            new[] { instance, GenerateOneBasedIndexExpression(expression.Arguments[0]) }));
+
+            case nameof(Array.GetUpperBound) when VersionAtLeast(7, 4):
+                return
+                    Coalesce(
+                        new SqlFunctionExpression(
+                            "array_upper",
+                            typeof(int),
+                            new[] { instance, GenerateOneBasedIndexExpression(expression.Arguments[0]) }));
+
+            default:
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Translates instance methods defined on <see cref="List{T}"/>.
+        /// </summary>
+        /// <param name="expression">The expression to translate.</param>
+        /// <returns>
+        /// A translated expression or null if no translation is supported.
+        /// </returns>
+        [CanBeNull]
+        Expression ListInstanceHandler([NotNull] MethodCallExpression expression)
+        {
+            if (!(expression.Object is Expression instance))
```

Comment:
Why this check? `MethodCallExpression.Object` already is an expression... If you just want a shorter local variable (`instance`) just assign...

(I think I wrote this elsewhere already)